scripting/shavit-timelimit.sp

```diff
@@ -244,6 +298,58 @@ void SetLimit(int time)
 	gCV_RestartGame.IntValue = 1;
 }
 
+public void Player_Spawn(Event event, const char[] name, bool dontBroadcast)
+{
+	int client = GetClientOfUserId(event.GetInt("userid"));
+
+	if(gB_ForceMapEnd)
+	{
+		CreateTimer(1.0, CheckRemainingTime, client, TIMER_REPEAT|TIMER_FLAG_NO_MAPCHANGE);
+	}
+}
+
+public Action CheckRemainingTime(Handle timer, int data)
+{
+	int client = GetClientFromSerial(data);
+	int timelimit;
```

Comment:
Every variable in bhoptimer starts initialized as a default value. If something needs to be fetched as a reference, initialize it as 0.